src/components/Api.js

```diff
@@ -0,0 +1,184 @@
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._cardID = cardID;
+  }
+
+  userProfileInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    });
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  editProfileRequest({ title, description }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: title,
+        about: description,
+      }),
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  //need to find a way to plug the data that i get from the add card form and push it into the addnewcards api call
+
+  addNewCards({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        link: link,
+      }),
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  confirmDeleteButton(cardID) {
+    return fetch(`${this._baseUrl}/cards/${cardID}`, {
+      method: "DELETE",
+      headers: this._headers,
+    });
+  }
+
+  //cardId in the like button request and the delete button request need to be updated with the cards id that comes from the server, so in order to do that we need so the request the id that comes from the server and imbed it into the requests when the action happens
+
+  /**
+   *
+   * @param {String} cardId needs to be replaced with the id that comes from the cards
+   */
+
+  likeButtonRequest(cardID) {
+    return fetch(`${this._baseUrl}/cards/${cardID}/likes`, {
+      method: "PUT",
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  removeLikeButtonRequest(cardID) {
+    return fetch(`${this._baseUrl}/cards/${cardID}/likes`, {
+      method: "DELETE",
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  updateProfilePictureRequest({ linkInput }) {
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar: linkInput,
+      }),
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return new Promise.reject(`Error : ${res.status}`);
+    });
+  }
+
+  async cardRenderer() {
```

Comment:
You don't need this method because you don't use it

Reply:
Im confused with this statement? thats my where my Promise.all statements lie.

Reply:
> Im confused with this statement? thats my where my Promise.all statements lie.

You didn't use `cardRenderer` from `Api`.


Now you have the correct method `getAppInfo` for promissing all. So, everything is ok now